python/cudf_polars/cudf_polars/callback.py

```diff
@@ -32,6 +33,37 @@
 __all__: list[str] = ["execute_with_cudf"]
 
 
+def enable_uvm(device: int):
+    managed_memory_is_supported = (
+        pylibcudf.utils._is_concurrent_managed_access_supported()
+    )
+    rmm_mode = "managed_pool" if managed_memory_is_supported else "pool"
```

Comment:
Unlike in cudf.pandas I think we want the fallback here to not be a normal pool but an async pool, which is the `default_memory_resource`.

Reply:
Sounds good. Made this change.